tools/fp20compiler/main.cpp

```diff
@@ -37,25 +37,69 @@ char * myin = 0;
 void translate(const char* s) {
     char* ns = strdup(s);
 
-    // printf("-- %s\b", s);
-
-    char* ts_start = strstr(ns, "!!TS1.0");
-    if (ts_start) {
-        char* ts_end = strstr(ts_start, "// End of program");
-        assert(ts_end);
-        *ts_end = 0;
-
-        ts10_init(ts_start);
-        ts10_parse();
-
-        *ts_end = '/';
-    }
-
-    char* rc_start = strstr(ns, "!!RC1.0");
-    if (rc_start) {
-        rc10_init(rc_start);
-        rc10_parse();
-    }
+    // Move to start of file
+    char* cursor = ns;
+
+    // Remember where the current shader starts
+    char* shader_start = NULL;
```

Comment:
This is misleading. It should say that we initially don't know where the current shader starts.

Reply:
I've rewritten the function by moving the search for the first shader out of the loop. It has more redundancy, but is easier to understand.